lib/spack/spack/installer.py

```diff
@@ -937,13 +960,12 @@ def __init__(
         # queue.
         if status == STATUS_REMOVED:
             raise InstallError(
-                f"Cannot create a build task for {self.pkg_id} with status '{status}'", pkg=pkg
+                f"Cannot create a task for {self.pkg_id} with status '{status}'", pkg=pkg
             )
-
         self.status = status
 
-        # Package is associated with a bootstrap compiler
-        self.compiler = compiler
+        # Getting the PID again because it will be needed for execute functionality.
```

Comment:
This comment isn't self-contained:
* Why "again"?  Where else?
* "execute functionality": which execute functionality?

```suggestion
        # save the PID, which is used by execute()
```